### PR TITLE
feat: implement TRY200

### PR DESCRIPTION
**Pull request by karpa4o4**
#2056

### CODE REVIEW

File: src/rules/tryceratops/rules/reraise_no_cause.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+/// TRY200
+pub fn reraise_no_cause(checker: &mut Checker, body: &[Stmt]) {
+    for stmt in body.iter() {
+        if let StmtKind::Raise { cause, .. } = &stmt.node {
+            if cause.is_none() {
+                checker.diagnostics.push(Diagnostic::new(
+                    ReraiseNoCause,
+                    Range::from_located(stmt),
+                ));
```
Comment (charliermarsh) [inline src/rules/tryceratops/rules/reraise_no_cause.rs:28]: I think we need to recurse here to handle cases like:

```py
def bad():
    try:
        process()
    except MyException:
        if True:
            raise MainFunctionFailed()
```

This came up in another PR -- here's my explanation: https://github.com/charliermarsh/ruff/pull/2073#discussion_r1083345058

Comment (charliermarsh) [reply src/rules/tryceratops/rules/reraise_no_cause.rs:28]: (Otherwise, looks good!)

**Review by charliermarsh — CHANGES_REQUESTED** [review]
Thanks! Think we just need to expand the logic to recurse over the statements.

**Comment by karpa4o4** [comment]
> Thanks! Think we just need to expand the logic to recurse over the statements.

Can you describe what statements you are talking about? If as in the screenshot, then it already works.
Screenshot_20230123_211848
Screenshot_20230123_211641

**Comment by charliermarsh** [comment]
Looks good, I'm gonna rebase then merge.

**Pull request merged**